test/insert.js

```diff
@@ -0,0 +1,49 @@
+var Pipeline = require( '..' )
+var assert = require( 'assert' )
+var fs = require( 'fs' )
+var path = require( 'path' )
+var Stream = require( 'stream' )
+
+describe( 'Pipeline.insert()', function() {
+
+  it( 'can insert a stream at specific indices', function( done ) {
```

Comment:
This test case doesn't seem to test that a non pass through stream is inserted correctly. Maybe we can insert transform streams that actually do something order-dependent to ensure it works correctly?